### PR TITLE
Fix Viewing Dag Code for Stateless Webserver

### PR DESCRIPTION
**Pull request by kaxil**
This bug was found by @KostyaEsmukov 

Details in https://github.com/apache/airflow/pull/8151#issuecomment-609439815

Basically when DagCode was instantiated it read the source code from Dag File, hence when the Webserver does not have access to the Dag File, it used to fail with FileNotFoundError:

```python
>>> DagCode(dag_orm.fileloc)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 4, in __init__
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/state.py", line 433, in _initialize_instance
    manager.dispatch.init_failure(self, args, kwargs)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py", line 69, in __exit__
    exc_value, with_traceback=exc_tb,
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 178, in raise_
    raise exception
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/state.py", line 430, in _initialize_instance
    return manager.original_init(*mixed[1:], **kwargs)
  File "/usr/local/lib/python3.7/site-packages/airflow/models/dagcode.py", line 59, in __init__
    self.source_code = DagCode._read_code(self.fileloc)
  File "/usr/local/lib/python3.7/site-packages/airflow/models/dagcode.py", line 63, in _read_code
    with open_maybe_zipped(fileloc, 'r') as source:
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/file.py", line 102, in open_maybe_zipped
    return io.open(fileloc, mode=mode)
FileNotFoundError: [Errno 2] No such file or directory: '<...my_dag...>'
```

This PR fixes this and adds additional tests to validate this behavior

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by kaxil** [comment]
Note: This PR is for v1-10-test branch to fix 1.10.10rc3 . I will port this to Master branch once we release 1.10.10

File: airflow/models/dagcode.py [hunk 1]
```diff
@@ -128,9 +122,9 @@ def bulk_sync_to_db(cls, filelocs, session=None):
                 os.path.getmtime(correct_maybe_zipped(fileloc)), tz=timezone.utc)
 
             if (file_modified - timedelta(seconds=120)) > old_version.last_updated:
-                orm_dag_code = DagCode(fileloc)
+                orm_dag_code = DagCode(fileloc, cls._get_code_from_db(fileloc))
```
Comment (mik-laj) [inline airflow/models/dagcode.py]: Is it necessary? This causes an n + 1 problem. Can you add tests for the number of queries for this method?

Comment (kaxil) [reply airflow/models/dagcode.py]: Fixed, I will add number of queries count when I PR against Master.

**Comment by kaxil** [comment]
cc @anitakar

File: airflow/models/dagcode.py [hunk 2]
```diff
@@ -169,27 +171,30 @@ def get_code_by_fileloc(cls, fileloc):
         :param fileloc: file path of a DAG
         :return: source code as string
         """
-        return DagCode(fileloc).code()
+        return cls.code(fileloc)
 
-    def code(self):
+    @classmethod
+    def code(cls, fileloc):
```
Comment (mik-laj) [inline airflow/models/dagcode.py:177]: I don't like this method name. It describes what he does. It does not contain any verb. However, this is not related to this PR. It should be `get_code_by_fileloc` or something similar.

**Review by mik-laj — APPROVED** [review]

File: airflow/models/dagcode.py (hunk 2 above)
Comment (kaxil) [reply airflow/models/dagcode.py:177]: I have a separate PR to improve the code in this file, that I have targetted against Master but yeah that is not the priority for now.
